src/main/java/eu/dissco/orchestration/backend/service/MappingService.java

```diff
@@ -30,8 +31,12 @@ public MappingRecord createMapping(Mapping mapping, String userId) throws Transf
     repository.createMapping(mappingRecord);
     return mappingRecord;
   }
-  public MappingRecord updateMapping(String id, Mapping mapping, String userId) {
-    var currentVersion = repository.getMapping(id);
+  public MappingRecord updateMapping(String id, Mapping mapping, String userId)
+      throws NotFoundException {
+    var currentVersion = repository.getMappingOmitDeleted(id);
+    if (currentVersion == null){
+      throw new NotFoundException("Requested mapping does not exist");
```

Comment:
Maybe at the id of the requested mapping in the exception. Could help debugging, if it goes wrong and we should have returned a mapping we can check the id with what is in the database.